src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -507,8 +523,8 @@ Error MeshForwarder::UpdateIp6RouteFtd(Ip6::Header &ip6Header, Message &aMessage
         }
         else if (aloc16 <= Mle::kAloc16DhcpAgentEnd)
         {
-            uint8_t contextId = static_cast<uint8_t>(aloc16 & Mle::kAloc16DhcpAgentMask);
-            SuccessOrExit(error = ContextIdRouteLookup(contextId, mMeshDest));
+            uint8_t contextId = static_cast<uint8_t>(aloc16 - Mle::kAloc16DhcpAgentStart + 1);
```

Comment:
I don't think `kAloc16DhcpAgentMask` is used anymore with this change, we can consider removing it?